Packages/com.unity.inputsystem/InputSystem/Editor/ProjectWideActions/ProjectWideActionsAsset.cs

```diff
@@ -92,6 +95,13 @@ internal static InputActionAsset CreateNewActionAsset()
 
             return asset;
         }
+        
+        private static void GetDefaultUIActionMap()
+        {
+            var json = File.ReadAllText(Path.Combine(Environment.CurrentDirectory, s_DefaultAssetPath));
```

Comment:
Given that the UI action map is cached into s_DefaultUIMap, wouldn't it make sense to move the check and early return into this function instead, e.g. have

```
if (s_DefaultUIMap != null) return s_DefaultUIMap;
var asset = GetOrCreate();
var json = ....
...
return s_DefaultUIMap;
```
inside this function? Another question would be, do we really need to cache it? It would occupy memory not really needed that often, so would suggest we could skip it? I think it would be fine loading it on demand when doing CheckForDefaultUIActionMapChanges()?

Reply:
If we skip caching we can also get rid of the GetDefaultUIActionMap() function and move that code into CheckForDefaultUIActionMapChanges()

Reply:
This line of of code should match the existing file read in `CreateNewActionAsset()` which is
`var json = File.ReadAllText(FileUtil.GetPhysicalPath(s_DefaultAssetPath));`

Additionally, as also mentioned in another comment, I think this line should be hoisted outside of this function and pass the json, so that there doesn't need to be 2 calls to this in `CreateNewActionAsset()`

`CheckForDefaultUIActionMapChanges()` will need this call added for that.

Reply:
Now I'm not caching the template/default UI action map as suggested by @ekcoh. Thanks, makes more sense now.
This means that for the JSON, the alternative is either to cache it, to avoid more than 1 reading in the Editor, or then we need to read the file close to when we need it. Is there any other solution you're seeing @jamesmcgill ?

Reply:
Will still need to be something like `var json = File.ReadAllText(FileUtil.GetPhysicalPath(path));` The `CurrentDirectory` way didn't work in Neturon/FPSSample.

Reply:
> Now I'm not caching the template/default UI action map as suggested by @ekcoh. Thanks, makes more sense now. This means that for the JSON, the alternative is either to cache it, to avoid more than 1 reading in the Editor, or then we need to read the file close to when we need it. Is there any other solution you're seeing @jamesmcgill ?

After your changes there is no longer a double file read, so not an issue anymore.

Reply:
@ekcoh I believe I made the changes you requested. Have a look :)